pkg/generate.go

```diff
@@ -56,15 +69,45 @@ func Generate(fs afero.Afero, gitOpsFolder string, outputFolder string, componen
 		deploymentFileName: deployment,
 	}
 
+	var service *corev1.Service
+	var route *routev1.Route
 	// If a targetPort was specified, also generate a service and route
-	if component.TargetPort != 0 {
-		service := generateService(component)
-		route := generateRoute(component)
+	if len(component.KubernetesResources.Services) == 0 && len(component.KubernetesResources.Routes) == 0 && component.TargetPort != 0 {
+		service = generateService(component)
+		route = generateRoute(component)
+	} else {
+		service, component.KubernetesResources.Services = &component.KubernetesResources.Services[0], component.KubernetesResources.Services[1:]
+		route, component.KubernetesResources.Routes = &component.KubernetesResources.Routes[0], component.KubernetesResources.Routes[1:]
+
+		var otherServices, otherRoutes []interface{}
+		for _, service := range component.KubernetesResources.Services {
+			otherServices = append(otherServices, service)
+		}
+		for _, route := range component.KubernetesResources.Routes {
+			otherRoutes = append(otherRoutes, route)
+		}
+
+		component.KubernetesResources.Others = append(component.KubernetesResources.Others, otherServices...)
+		component.KubernetesResources.Others = append(component.KubernetesResources.Others, otherRoutes...)
+	}
```

Comment:
Looks like you are picking out the first service and route and then putting the rest under Others.  What's the reason for this?

Reply:
its just to organize things nicely. In the main branch we have one resource in deployment.yaml, route.yaml, etc.. So if there are N deployments, services, routes, etc.; we take the first one and put the rest in a separate file.

Check here on how this looks: https://github.com/maysunfaisal/petclinic-lK5Sb-wash-touch/tree/main/components/component-sample-2/base

Reply:
Is it better to use one file per manifest?  I'm wondering if in the future, if there are other tools that 'analyze' the application (or the GitOps repo contents), they can more easily check the individual files.  For example, if the tool wants to know the number of services in the application, they can check the filenames that have the word `service` in them, presuming, that is the naming convention (eg. 100-service.yaml, 101-service.yaml, etc.), or put them in separate folders.  This directory layout is something that is opinionated in GitOps.  Other tools use completely different layouts.

Reply:
@keithchong so if we have N services, we need to have N service.yaml files?

Reply:
@maysunfaisal , yes. Have you talked to others about this design?

Reply:
@keithchong yes, I have. I will get back to this PR shortly and post comments about that discussion.

Reply:
Ok, the proposal to iterate on this is acceptable.  We can open an issue for this.

Yes, I've seen an example where there were 1000 folders.  Perhaps the API should allow the client to choose what structure they want.

For a custom resource, we'll have to come up with a convention.

Could you elaborate on the third point?  The custom resource has a file name that clashes?

Reply:
@keithchong where do we track issues for this repo, it is a question we have not discussed before! Does GitOps have a JIRA that perhaps can be used? I can create an issue there with what we discussed here.

For the third point, lets say this module generated 100-service.yaml; but a user has manually gone to the repo and committed 101-service.yaml. This module now cannot generate the next file as 101-service.yaml for instance.. So we need to take care of such clashes and coming up with a good logic and convention.

Reply:
Generating 'from scratch' will not be problem.  If the repo already exists, then I think that when you write out, then it must check first if the file with that name already exists and then react accordingly?  We'll have to be careful and consider various scenarios.

Reply:
LGTM with caveat that you'll open an issue in this repo for the enhancement.

Reply:
@keithchong I have opened https://github.com/redhat-developer/gitops-generator/issues/35 as we discussed